weld/type_inference.rs

```diff
@@ -167,9 +167,9 @@ fn infer_locally(expr: &mut PartialExpr, env: &mut TypeMap) -> WeldResult<bool>
                                 .zip(vec![key_type.clone(), value_type.clone()].iter_mut()) {
                             changed |= try!(push_type(field_ty, child_expr_field, "ToVec"));
                         }
-                    } else {
-                        return weld_err!("Internal error: toVec argument needs to be a dict");
-                    }
+                    }//  else {
```

Comment:
The type inference fails for the following kind of expression
`let rs = result(for(x, {dictmerger[i32,i32,+], dictmerger[i64,i64,+]}, |b,i,e| {b.$0, b.$1}))
  let agg_dict = result(rs.$0)
`

Here agg_dict has an "Unkown" type and so we can't call tovec on it. 
I had to comment this else statement out so I could proceed with generating the code.
There's probably a more elegant solution.

Reply:
This might actually fixed with shoumik's recent change fixing type inference on lets.